pyproj/database.pyx

```diff
@@ -483,3 +488,105 @@ def get_database_metadata(str key not None):
         return metadata
     finally:
         pyproj_context_destroy(context)
+
+
+def query_geodetic_crs_from_datum(
+    str crs_auth_name,
+    str datum_auth_name not None,
+    str datum_code not None,
+    pj_type=None
+    ):
+    """
+    .. versionadded:: 3.7.0
+
+    Return GeodeticCRS that use the specified datum
+
+    See: :c:func:`proj_query_geodetic_crs_from_datum`
+
+    Parameters
+    ----------
+    crs_auth_name: str | None
+        The authority name to filter by (e.g. EPSG, ESRI). None is all.
+    datum_auth_name: str
+        The authority of the datum
+    datum_code: str
+        Datum code
+    pj_type: pyproj.enums.PJType | None, optional
+        The type of object to get the CRSs. Can be PJType.GEOCENTRIC_CRS,
+        PJType.GEOGRAPHIC_3D_CRS, PJType.GEOGRAPHIC_2D_CRS or None for all.
+
+    Returns
+    -------
+    list[_CRS]
+    """
+
+    cdef const char* c_crs_type = NULL
+    cdef bytes b_crs_type
+    cdef str crs_type = ""
+    if pj_type is None:
```

Comment:
```python
 cdef PJ_TYPE cpj_type = get_pj_type(pj_type)
```

Reply:
I do not understand this suggestion. If the user sets `None`, then c_crs_type is NULL, and the C function will return without filtering by type.

Reply:
The suggestiin is not in the exact location. It would be applied after that.

Reply:
I think I understood what you mean. I added it, and some tests.